src/main/kotlin/org/rust/ide/sdk/add/RsSdkPathChoosingComboBox.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.sdk.add
+
+import com.intellij.openapi.application.AppUIExecutor
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.ModalityState
+import com.intellij.openapi.fileChooser.FileChooser
+import com.intellij.openapi.projectRoots.Sdk
+import com.intellij.openapi.ui.ComboBoxWithWidePopup
+import com.intellij.openapi.ui.ComponentWithBrowseButton
+import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.ui.AnimatedIcon
+import com.intellij.ui.ComboboxSpeedSearch
+import com.intellij.ui.components.fields.ExtendableTextComponent
+import com.intellij.ui.components.fields.ExtendableTextField
+import com.intellij.util.PathUtil
+import org.rust.ide.sdk.RsDetectedSdk
+import org.rust.ide.sdk.RsSdkListCellRenderer
+import org.rust.ide.sdk.RsSdkType
+import org.rust.ide.sdk.add.RsSdkPathChoosingComboBox.Companion.addToolchainsAsync
+import javax.swing.plaf.basic.BasicComboBoxEditor
+
+/**
+ * A combobox with browse button for choosing a path to SDK, also capable of showing progress indicator.
```

Comment:
Fixed. But now `browse button` opens Rust toolchain list. It is useful since CLion doesn't provide a way to add/delete/edit SDKs.

![image](https://user-images.githubusercontent.com/6079006/95056146-f82a9e80-06fc-11eb-8925-9e41022cee78.png)